lib/vault/api/logical/versioned.rb

```diff
@@ -0,0 +1,86 @@
+require_relative "unversioned"
+require_relative "../secret"
+require_relative "../../client"
+require_relative "../../request"
+require_relative "../../response"
+
+module Vault
+  module Logical
+    class Versioned < Unversioned
+      # List the secrets at the given path, if the path supports listing. If the
+      # the path does not exist, an exception will be raised.
+      #
+      # @example
+      #   Vault.logical.list("secret") #=> [#<Vault::Secret>, #<Vault::Secret>, ...]
+      #
+      # @param [String] path
+      #   the path to list
+      #
+      # @return [Array<String>]
+      def list(mount, path = "", options = {})
+        headers = extract_headers!(options)
+        json = client.list("/v1/#{mount}/metadata/#{encode_path(path)}", {}, headers)
+        json[:data][:keys] || []
+      rescue HTTPError => e
+        return [] if e.code == 404
+        raise
+      end
+
+      # Read the secret at the given path. If the secret does not exist, +nil+
+      # will be returned.
+      #
+      # @example
+      #   Vault.logical.read("secret/password") #=> #<Vault::Secret lease_id="">
```

Comment:
This documentation needs to be updated right? For example this should be something like `read("secret", "password")`, correct?

Reply:
Good catch, yes. If the general approach of this PR is fine and expected to be merged, then I'm very happy to add a commit updating the inline docs and the README.

Reply:
👍 Thanks a lot for working on this, much appreciated!

Reply:
Just improved the inline docs a bit :)